### DISCUSSION TITLE
Revealjs: mix horizontal and vertical slides in teaching context

### DISCUSSION THREAD

**Discussion by cathblatter** [quarto-dev/quarto-cli#826] [Q&A] [answered]
A question for using revealjs-slides in a teaching context with conditional inclusion/exclusion of certain slides - my questions is somewhat related to this (https://github.com/quarto-dev/quarto-cli/discussions/373) but targets rather the output *content* than the format and I am looking for feedback:

_Current workflow:_
I use {xaringan} for teaching: e.g. 3 slides with theory, then 1 slide with questions and then a last slide with the answers to the questions before. I would render the html-slides and then use {xaringanBuilder::build_pdf()} to build the slides to share with students pre-class, but manually delete the slides with the answers in Acrobat before sharing. Obvs not very elegant.

_Idea:_
With quarto I thought I can use horizontal slides as the main content (3 slides with theory, 1 slide with questions) and the vertical slides with answers. I could then pick between option e.g. "teach" to render all slides for html-sharing during class and option "student" to only render the horizontal slides and subsequently convert those to pdf for sharing with students.

Trying this out I had several thoughts:

- Is it correct that horizontal/vertical slides only work with Level 1 / Level 2 headings? I would be interested in having all headers the same ( `##`). Is there a way to add a something like a class e.g. `{.answer-vertical}` manually to a slide?

- I could add a custom lua-filter (e.g. #373) but instead of format (that is implemented now here: https://github.com/quarto-dev/quarto-cli/issues/646#issuecomment-1115459837) add a custom class based on the content/target {.answer-vertical} to reset this slide when wanting to render for students.

Does it make sense to use horizontal/vertical slides that way? Is the 'content-wise' showing of slides dependent on audience (not format) on the roadmap? Or is there an easier way to this situation that I am currently totally blind to?

Thanks!

**Comment by jjallaire** [comment, marked as answer]
Any side that you mark with `visibility="hidden"` will not render (see https://quarto.org/docs/presentations/revealjs/advanced.html#slide-visibility)

So you could write a simple Lua filter that adds this attribute to all slides with a given class (e.g. `.answer`). For example (not tested!):

```lua
function Heading(el)
  if (el.attr.classes:find("answer")) then
    el.attr.attributes["visibility"] = "hidden"
    return el
  end
end
```

Then just only apply the filter when you are creating the student slides.
